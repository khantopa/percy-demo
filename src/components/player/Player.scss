@import '../../styles/main.scss';

.player {
  height: 200px;
  width: 200px;
  animation: fadeIn 2s ease;
  &_type {
    text-align: center;
    font-size: 16px;
  }
}

@media (max-width: $mobile-width) {
  .player {
    width: 130px;
    height: 130px;
    &_type {
      font-size: 14px;
    }
  }
}