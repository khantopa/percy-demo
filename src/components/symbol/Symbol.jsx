import React from 'react';
import PropTypes from 'prop-types';

import './Symbol.scss';


const Symbol =(props) => {
  const { symbol, color } = props;
  const imgUrl = `../../../assets/${symbol}.png`;
  const style = {
    backgroundColor: color,
    backgroundImage: `url(${imgUrl})`,
  }
  return <div className="symbol" style = { style } ></div>
}


Symbol.defaultProps = {
  symbol: "",
  color: "#D8EEF2"
};


Symbol.propTypes = {
  color: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired
}


export default Symbol;


