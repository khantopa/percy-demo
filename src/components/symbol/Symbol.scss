@import '../../styles/main.scss';


.symbol { 
  box-sizing: border-box;
  background: $primary-light center/60% no-repeat;
  height: 100%;
  width: 100%;
  border: 3px solid $light;
  border-radius: 50%;
  box-shadow: inset 0 5px 10px 4px rgba(0, 0, 0, .35);
}