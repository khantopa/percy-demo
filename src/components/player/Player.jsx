import React from 'react';
import PropTypes from 'prop-types';

import './Player.scss';

import Symbol from '../symbol/Symbol';


const Player = (props) => {

  const player = props.player;
  return (
    <div className="player">
      <h5 className="player_type">
        { player.name }: &nbsp;
        <span className="player_score"> { player.score }  </span>
      </h5>
      <Symbol symbol= { player.symbol } />
    </div>
  );
}

Player.propTypes = {
  player: PropTypes.object.isRequired
}

export default Player;

