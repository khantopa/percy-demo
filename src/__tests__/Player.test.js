import Player from '../components/player/Player';
import Symbol from '../components/symbol/Symbol';

let wrapper, props;


beforeEach(() => {
  props = {} ; 
  wrapper =shallow(<Player player = { props } />);
});

describe('<Player>', () => {

  it('should <Board> defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders <Player> without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('render <Symbol> without crashing', () => {
    expect(wrapper.find(Symbol)).toHaveLength(1);
  });

});
