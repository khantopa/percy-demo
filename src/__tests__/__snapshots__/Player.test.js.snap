// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player> renders <Player> without crashing 1`] = `
<div
  className="player"
>
  <h5
    className="player_type"
  >
    : Â 
    <span
      className="player_score"
    >
       
        
    </span>
  </h5>
  <Symbol
    color="#D8EEF2"
    symbol=""
  />
</div>
`;
